docs/turn-howto.md

```diff
@@ -129,26 +144,84 @@ This will install and start a systemd service called `coturn`.
     We recommend that you only try to set up TLS/DTLS once you have set up a
     basic installation and got it working.
 
+    NB: If your TLS certificate was provided by Let's Encrypt, TLS/DTLS will
+    not work with any Matrix client that uses Chromium's WebRTC library (such
+    as Element Android). See [this case](https://github.com/vector-im/element-android/issues/1533) for more details.
```

Comment:
I'm struggling to believe this is true.  Matrix.org's TLS cert is provided by LE, and it seems to work? Doesn't Google Chrome use the same WebRTC library?

The issue just says "socket buffer operation error", which doesn't really tell me a lot. I have no idea where that error appears or what its effect is.

Reply:
Have to agree, LP.net uses LE and it seems to work without complaint.

Reply:
Yes, there can indeed be connectivity issues when using one of the mobile Element clients. The same issue affects the Jitsi Meet app on Android.

I'll add links to the Element iOS & Jitsi Meet issues:
https://github.com/vector-im/element-ios/issues/2712
https://github.com/jitsi/jitsi-meet/issues/6383#issuecomment-632659991

From my testing, calls fail to connect only if at least one client is Element Android, call traffic is required to go through the TURN server, and only `turns` connections are available. (If unencrypted TURN URIs are available, they continue to work as fallbacks.)

To test forcing traffic through the TURN server (as opposed to using direct p2p connections, which AIUI is what STUN tries to set up--I'm still foggy on the full details of this), I made calls to/from a work laptop on a corporate VPN. To ensure encryption was used, I either disabled all `turn:` entries from my homeserver's `turn_uris`, or I left them enabled, but blocked traffic to/from ports 3478-3479 for my laptop & (Android) mobile, with either device serving as the other end of a call with the work laptop.

The results of this were:
* Calls between two web browsers (personal & work laptops) were successful, whether or not `turn:` URIs were disabled, or ports 3478-3479 were blocked. Running `tcpdump` on my personal laptop showed traffic flowing through port 5349.
* Calls between my phone & the work laptop failed unless I enabled `turn:` URIs or unblocked the ports.

Critically, the results were identical even when making calls from a matrix.org account, and thus while using its TURN server & its certificate.

(If it makes a difference, turn.matrix.org's `turns:` URI uses TCP on port 443, not 5349--but if anything, that should help connections be more reliable, as no firewall should block TLS traffic on port 443.)

Results were similar when testing my self-hosted Jitsi: calls from the Android app were unable to get p2p connections with `turns`, and instead fell back to using the videobridge, while calls between two web browsers could always use p2p.

(FWIW, Jitsi Meet is what informed me of the possibility of non-TURN p2p connections: calls between my personal laptop & my phone on mobile data get tagged as "(p2p)", whereas calls involving the work laptop behind its VPN were labelled as "(p2p)(turn)", unless the videobridge was used as a fallback. Any setup that gets marked as "(p2p)" in Jitsi Meet always results in a successful call in Element.)

Reply:
ookay, thanks for the detailed information, but could you put it all on the relevant issue? (https://github.com/vector-im/element-android/issues/1533, I guess.) It's valuable research, which is going to get lost in this PR thread, and it needs to be reported to the people who maintain the software with the bugs!

(aside: I note that https://github.com/vector-im/element-ios/issues/2712#issuecomment-869274355 says it's fine if you correctly publish the intermediate certificates - is that not correct)?

We can consider adding a warning to `turn-howto.md`, but it's an orthogonal problem to NAT - please could you remove it from this PR for now?

Reply:
Removed, and commented in https://github.com/vector-im/element-android/issues/1533#issuecomment-996409199